File: SwiftAutomataLibrary/Transition.m
#import <Foundation/Foundation.h>

#include "Transition.h"

@implementation Transition_objc

-(instancetype)init: (NSString *) fromState toState:(NSString *) toState symbols:(NSArray *) symbols isEpsilonIncluded:(bool) isEpsilonIncluded {
    self = [super init];
    self->_fromState = fromState;
    self->_toState = toState;
    self->_symbols = symbols;
    self->_isEpsilonIncluded = isEpsilonIncluded;
    return self;
}

@end

File: SwiftAutomataLibrary/Transition.swift
public struct Transition {
    public let fromState: String
    public let toState: String
    public let symbols: [String]
    public let isEpsilonIncluded: Bool
    
    public init(
        fromState: String,
        toState: String,
        symbols: [String],
        isEpsilonIncluded: Bool
    ) {
        self.fromState = fromState
        self.toState = toState
        self.symbols = symbols
        self.isEpsilonIncluded = isEpsilonIncluded
    }
}
